src/cluster.js

```diff
@@ -32,18 +32,17 @@ const is = require('is');
  * const cluster = instance.cluster('my-cluster');
  */
 class Cluster {
-  constructor(instance, name) {
+  constructor(instance, id) {
     this.bigtable = instance.bigtable;
     this.instance = instance;
 
-    let id = name;
-
-    if (!id.includes('/')) {
-      id = `${instance.name}/clusters/${name}`;
+    if (id.indexOf('/') > -1) {
+      throw new Error(
+        'Cluster Id is not formatted correclty. Just `mycluster` rather than `projects/myproject/instances/myinstance/clusters/mycluster`'
```

Comment:
Across our different APIs, we accept the dual format. If the user is over-specific, what's the harm?

Reply:
One way or another, there needs to be error checking.  We'd have to check that `id.contains(instance.name)` here.

Reply:
Perhaps error prone, a lot more room to make a typo while spelling out
`"projects/myproject/instances/myinstance/clusters/mycluster"`
rather then just passing `"mycluster"` and let the client convert it into fully qualified name?

Reply:
That's a fine idea. Allowing the upstream API to communicate the error at the time of an operation is usually an okay fallback. But in this case, I would agree that since you have the `instance.name` available, you might as well catch the error asap.

Reply:
The user should be allowed to provide any input that we can make sense of. We can handle both the name or ID. We can encourage the easy one, but we shouldn't penalize the user for being verbose.

Reply:
Ack. We'll change the logic to include a correctness check that makes sure that the parent and full name are in sync.